de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/security/impl/UserRolePermissionsDAO.java

```diff
@@ -231,7 +233,37 @@ public OrgPermissions retrieveOrgPermissions(final I_AD_Role role, final int adU
 		}
 		else
 		{
-			return retrieveRoleOrgPermissions(role.getAD_Role_ID(), adTreeOrgId);
+	
+			if (role.isAccessAllOrgs())
+			{
+				final int AD_Client_ID = role.getAD_Client_ID();
+				final Properties ctx = InterfaceWrapperHelper.getCtx(role);
+
+				//
+				// if role has acces all org, then behave as would be * access
+				final OrgPermissions.Builder builder = OrgPermissions.builder()
+						.setOrg_Tree_ID(adTreeOrgId);
+				
+				// org *
+				OrgResource resource = OrgResource.of(AD_Client_ID, 0);
```

Comment:
also pls use org.compiere.util.Env.CTXVALUE_AD_Org_ID_System instead of hardcoded "0".